cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/CassandraDataLayer.java

```diff
@@ -338,11 +338,13 @@ else if (clearSnapshotStrategy.hasTTL())
     private CompletionStage<Map<String, AvailabilityHint>> createSnapshot(ClientConfig options, RingResponse ring)
     {
         Map<String, PartitionedDataLayer.AvailabilityHint> availabilityHints = new ConcurrentHashMap<>(ring.size());
+        Map<String, Boolean> uniqueInstances = new ConcurrentHashMap<>();
 
         // Fire off create snapshot request across the entire cluster
         List<CompletableFuture<Void>> futures =
         ring.stream()
             .filter(ringEntry -> datacenter == null || datacenter.equals(ringEntry.datacenter()))
+            .filter(ringEntry -> uniqueInstances.putIfAbsent(ringEntry.fqdn(), true) == null)
```

Comment:
Does this assume that the `RingResponse` now returns an entry per vnode?

Reply:
Yes, the RingResponse already contains an entry per vnode and I've confirmed this locally - Will also add a test.

Reply:
To confirm, RingResponse comes from calling the keyspace specific ring endpoint on the sidecar, which returns an entry per token range rather than per instance.

For a 6 node cluster with 4 num_tokens on each node:
```
> curl -s $(hostname):7001/api/v1/cassandra/ring | jq '. | length'
24
> curl -s $(hostname):7001/api/v1/cassandra/ring | jq '. | group_by(.address) | length'
6
```